aws_lambda_powertools/metrics/provider/base.py

```diff
@@ -2,19 +2,17 @@
 
 import functools
 import logging
+from abc import ABC, abstractmethod
 from typing import Any, Callable, Dict, Optional
 
-from typing_extensions import Protocol, runtime_checkable
+from aws_lambda_powertools.metrics.provider import cold_start
 
 logger = logging.getLogger(__name__)
 
-is_cold_start = True
 
-
-@runtime_checkable
-class MetricsProviderBase(Protocol):
+class BaseProvider(ABC):
     """
-    Interface for MetricsProvider.
+    Class for metric provider interface.
 
     This class serves as an interface for creating your own metric provider. Inherit from this class
     and implement the required methods to define your specific metric provider.
```

Comment:
```suggestion
    Interface to create a metrics provider.

    BaseProvider implements `log_metrics` decorator for every provider as a value add feature.
```

We can keep it short here since there was duplicate. Inform whoever is implementing it that `log_metrics` is a value add unless they want to override.

If the value add features grow, we can make into a list later.